app/controllers/items_controller.rb

```diff
@@ -5,23 +5,32 @@ class ItemsController < ApplicationController
   #before_action :current_user.admin?         #WIP must verify that the user is an admin in order for to run any procudure in this controller. An admin
                                               #should be the only user that is able to add/edit/delete items in the database
   
+
   def index                                   #items in library
+    if not params[:filter].nil? and params[:filter]!=""
```

Comment:
`if not params[:filter].nil?` can be written as `if params[:filter]` since anything non-nil evaluates to `true`; same applies to other `if not nil` statements below.

Reply:
Also just realized I think you're looking for `params[:filter].present?` here (`present?` checks that something isn't nil or an empty string)